homeassistant/components/bosch_shc/config_flow.py

```diff
@@ -0,0 +1,220 @@
+"""Config flow for Bosch Smart Home Controller integration."""
+import logging
+from os import makedirs
+
+from boschshcpy import SHCRegisterClient, SHCSession
+from boschshcpy.exceptions import (
+    SHCAuthenticationError,
+    SHCConnectionError,
+    SHCmDNSError,
+    SHCRegistrationError,
+    SHCSessionError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.components.zeroconf import async_get_instance
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_TOKEN
+
+from .const import (
+    CONF_HOSTNAME,
+    CONF_SHC_CERT,
+    CONF_SHC_KEY,
+    CONF_SSL_CERTIFICATE,
+    CONF_SSL_KEY,
+)
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+HOST_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, host, cert, key):
```

Comment:
Please type the functions completely or not at all.
```suggestion
async def validate_input(hass: core.HomeAssistant, host: str, cert: str, key: str) -> None:
```